src/internal/connector/exchange/service_restore.go

```diff
@@ -649,10 +649,17 @@ func establishContactsRestoreLocation(
 		return cached, nil
 	}
 
+	folders[0] = "Corso_Restore_02-May-2023_00:06:19"
 	ctx = clues.Add(ctx, "is_new_cache", isNewCache)
 
+	// This is where destination folder gets created by corso in graph
+	// This is the POST operation which may fail if the folder exists
 	temp, err := ac.Contacts().CreateContactFolder(ctx, user, folders[0])
 	if err != nil {
+		// TODO:: Add status code check too
```

Comment:
Status code checks shouldn't be necessary, since the graph client transforms status codes into error interfaces.  It's only in explicit corner-case conditions (eg: using a non-graph http client, or catching an error graph doesn't handle) that we compare status codes.